src/twisted/python/reactor_unittest.py

```diff
@@ -0,0 +1,584 @@
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+from unittest import TestCase
+
+from twisted.internet import reactor
+from twisted.internet._signals import _SIGCHLDWaker, _SocketWaker, _UnixWaker
+from twisted.internet.defer import Deferred
+from twisted.python.failure import Failure
+
+
+def benchmarkWithReactor():
+    """
+    Decorator for running the test with the benchmark.
+    """
+
+    def decorator(test_target):
+        """
+        This is the actual decorator.
+        """
+
+        def benchmark_test(self):
+            def run_test():
+                self.getDeferredResult(test_target(self))
+
+            self.benchmark(run_test)
+
+        return benchmark_test
+
+    return decorator
+
+
+class ReactorTestCase(TestCase):
+    """
+    Standard library unittest test case that can spin the reactor.
+
+    Provides support for running deferred and start/stop the reactor during
+    tests.
+    """
+
+    # Number of second to wait for a deferred to have a result.
+    DEFERRED_TIMEOUT = 1
+
+    # List of names for delayed calls which should not be considered as
+    # required to wait for them when running the reactor.
+    EXCEPTED_DELAYED_CALLS = []
+
+    EXCEPTED_READERS = [
+        _UnixWaker,
+        _SocketWaker,
+        _SIGCHLDWaker,
+    ]
+
+    # Scheduled event to stop waiting for a deferred.
+    _reactor_timeout_call = None
+
+    def setUp(self):
+        super(ReactorTestCase, self).setUp()
+        self._timeout_reached = False
+        self._reactor_timeout_failure = None
+
+    def tearDown(self):
```

Comment:
This does its best to clean the reactor... but is not 100% reliable

I think that this should be good enough for the benchmark tests.

If a benchmark test fails, it might cause side-effects and trigger other tests got fail.

but the idea is that the benchark tests should all succeed